src/calling/haplotype_abundances.rs

```diff
@@ -1,43 +1,38 @@
+use anyhow::Result;
+use derive_builder::Builder;
 use hdf5;
+use kernel_density;
 use rust_htslib::bcf;
-use derive_builder::Builder;
-use anyhow::Result;
-use std::path::Path;
+
 
 #[derive(Builder)]
 #[builder(pattern = "owned")]
-pub(crate) struct Caller<'a> {
-    hdf5_reader: hdf5::Reader<'a>,
+pub(crate) struct Caller {
+    hdf5_reader: hdf5::File,
     vcf_reader: bcf::Reader,
 }
 
-impl<'a> Caller<'a> {
+impl Caller {
+    pub(crate) fn cdf(&self) -> Result<()> {
+        let hdf5 = &self.hdf5_reader;
+        let group = hdf5.group("bootstrap")?;
+        let num_bootstraps = hdf5.dataset("/aux/num_bootstrap")?.read_scalar()?;
+        let mut bootstraps = Vec::new();
+        for i in 0..num_bootstraps {
+            let dataset = group.dataset(&format!("bs{i}", i = i))?;
+            let tpm = dataset.read_1d::<u64>()?;
+            bootstraps.extend(tpm.to_vec());
+        }
+        let n_values = hdf5.dataset("/aux/lengths")?.read_scalar()?;
+        let array = ndarray::Array2::from_shape_vec((num_bootstraps, n_values), bootstraps)?;
+        let array = array.t();
+        let cdf: Vec<_> = array
```

Comment:
It would be good to not only return the CDF, but also via a struct a second field containing the raw bootstrap values. This way we can plot them for QC. THis would also be the next task: add a command line flag to your subcommand that instructs varlociraptor to store plots for each quantified haplotype in a given directory. Such plotting is in rust best done via [vega-lite](https://vega.github.io/vega-lite). @jafors is an expert in that. Basically, vega-lite is a way to define a plot along with its data in JSON format. You basically just store that format from rust, and then subsequent rules in a workflow can render that to an SVG or PDF file using the vega-lite command line utility.

Regarding the plot:
x-axis: abundance
y-axis: density

The plot should superimpose the CDF with a histogram of the bootstrap values.